### ISSUE TITLE
Why doesn't `subscribe` dispatch the current state?

### ISSUE THREAD

**Issue by mAAdhaTTah** [reactjs/redux#1707]
I'm working on integrating `redux` into a project, and I'm setting up all of my view components (not using React here) to get passed the `state` and render/update themselves based on it. I know this is a fairly minor issue, but I'd like to know why the `state` object isn't passed an argument to the `subscribe` callback.

For example, what I'd like to do is this:

```js
const store = createStore(reducer, state);
store.subscribe(view.render);
```

But what I have to do is this:

```js
const store = createStore(reducer, state);
store.subscribe(() => {
    view.render(store.getState());
});
```

Is there a reason for this?

**Comment by rossipedia** [comment]
This has been discussed previously: 

#1513 
#303

TL;DR: Being notified that a store has changed is orthogonal to getting the current state. Read the above referenced issues for more details.

**Comment by mAAdhaTTah** [comment]
Fair enough, thanks for pointing me in the right direction.

**Comment by gaearon** [comment]
FWIW we can finally allow this after #1702 because store enhancers are exposed to a subset of API without `subscribe`.
